proposals/0004.md

```diff
@@ -0,0 +1,153 @@
+* Title: *Radicle client URI scheme*
+* Champion: *@rudolfs, rudolfs@osins.org*
+* Conversation:
+  * [Radicle Community discussion][community]
+  * [Shareable identifier discussion][shareable]
+  * [Deep linking discussion][deeplink]
+* References:
+  * [List of URI schemes][urilist]
+
+# Motivation
+
+We have received multiple feature requests from our community regarding
+deep-linking in our client application. A common way to enable such
+functionality is to devise a URI scheme along with a custom protocol and
+implement a handler which listens for requests on the given protocol in a
+client application.
+
+Given that there might be multiple competing Radicle client implementations and
+to make sure that shared URIs are compatible with as many Radicle client
+implementations as possible we decided to have a specification for our URI
+scheme.
+
+
+# Explanation
+
+The main use case for a custom protocol is to be able to open a Radicle client
+application by clicking on a link on the web in an external browser. The
+Radicle client should be able to recognise what kind of resource is being
+requested and be able to render the request accordingly.
+
+Besides showing resources it might be valuable to perform certain actions on
+said resources. For example a user may be able to start tracking a project by
+clicking a link or initiate a fund transfer to a certain address.
+
+
+# Recommendation
+
+To make implementation easier we want to use existing URI parsing libraries,
+therefore our URI scheme should be parsable according to [RFC 3986][rfc3986].
+
+To make sure our URI scheme is extensible in the future and clients can provide
+backwards compatibility for older links we should have a version identifier in
+the scheme.
+
+The URI scheme to be used by Radicle client applications should adhere to the
+following scheme:
+
+    radicle://[<protocolVersion>/<context>/<entity>?action=<action>]
+
+_Note:_ brackets denote optionality.
+
+Where:
+
+- `radicle:` is the Radicle client protocol name.
+- `<protocolVersion>` is a version number of the Radicle client protocol. This
+  should be the letter `v` followed by a positive integer. The current value
+  is: `v0`.
+- `<context>` is a namespace to distinguish which component of the client
+  application should handle the given request. Supported values:
+  - `link` for code collaboration functionality
+  - `ethereum` for Ethereum functionality
+- `<entity>` can be one of the following, depending on `<context>`:
+  - a `Radicle ID`
+  - an Ethereum address or
+  - an Ethereum transaction.
+- `<action>` is the action that will be performed:
+  - `Radicle ID`s support `show` and `track` actions
```

Comment:
Couldn't we also have `track` a `PeerId` for a `Urn`? :) But maybe enumerating _all_ the use cases isn't up to this RFC, but could be worth documenting in Upstream for what cases we could potentially have and which are implemented.

Reply:
I also think it's out of scope for this PR, we can always come back and update the spec.